pmm-admin_test.go

```diff
@@ -1299,6 +1305,352 @@ func testAddLinuxMetricsWithAdditionalArgsFail(t *testing.T, data pmmAdminData)
+	assert.Nil(t, err)
+	expected := `\[linux:metrics\] OK, now monitoring this system.
+\[mysql:metrics\] OK, now monitoring MySQL metrics using DSN root:\*\*\*@tcp\(127.0.0.1:3306\)
+\[mysql:queries\] OK, now monitoring MySQL queries from perfschema using DSN root:\*\*\*@tcp\(127.0.0.1:3306\)
+`
+	assertRegexpLines(t, expected, string(output))
+}
+
+func testAddMySQLWithDisableSlowLogsRotation(t *testing.T, data pmmAdminData) {
+	defer func() {
+		err := os.RemoveAll(data.rootDir)
+		assert.Nil(t, err)
+	}()
+
+	os.MkdirAll(data.rootDir+pmm.PMMBaseDir, 0777)
+	os.MkdirAll(data.rootDir+pmm.AgentBaseDir+"/bin", 0777)
+	os.MkdirAll(data.rootDir+pmm.AgentBaseDir+"/config", 0777)
+	os.MkdirAll(data.rootDir+pmm.AgentBaseDir+"/instance", 0777)
+	os.Create(data.rootDir + pmm.PMMBaseDir + "/node_exporter")
+	os.Create(data.rootDir + pmm.PMMBaseDir + "/mysqld_exporter")
+	os.Create(data.rootDir + pmm.PMMBaseDir + "/mongodb_exporter")
+	os.Create(data.rootDir + pmm.PMMBaseDir + "/proxysql_exporter")
+	os.Create(data.rootDir + pmm.AgentBaseDir + "/bin/percona-qan-agent")
+
+	f, _ := os.Create(data.rootDir + pmm.AgentBaseDir + "/bin/percona-qan-agent-installer")
+	f.WriteString("#!/bin/sh\n")
+	f.WriteString("echo 'it works'")
+	f.Close()
+	os.Chmod(data.rootDir+pmm.AgentBaseDir+"/bin/percona-qan-agent-installer", 0777)
+
+	f, _ = os.Create(data.rootDir + pmm.AgentBaseDir + "/config/agent.conf")
+	f.WriteString(`{"UUID":"42","ApiHostname":"somehostname","ApiPath":"/qan-api","ServerUser":"pmm"}`)
+	f.WriteString("\n")
+	f.Close()
+	os.Chmod(data.rootDir+pmm.AgentBaseDir+"/bin/percona-qan-agent-installer", 0777)
+	config := pc.QAN{}
+	in := &proto.Instance{
+		Subsystem: "mysql",
+		UUID:      "13",
+	}
+	// Create fake api server.
+	{
+		fapi := fakeapi.New()
+		fapi.AppendRoot()
+		fapi.AppendQanAPIPing()
+		fapi.AppendConsulV1StatusLeader()
+		node := api.CatalogNode{
+			Node: &api.Node{},
+		}
+		clientName, _ := os.Hostname()
+		fapi.AppendConsulV1CatalogNode(clientName, node)
+		fapi.AppendConsulV1CatalogService()
+		fapi.AppendConsulV1CatalogRegister()
+		in := &proto.Instance{
+			Subsystem: "mysql",
+			UUID:      "13",
+		}
+		agentInstance := &proto.Instance{
+			Subsystem: "agent",
+			UUID:      "42",
+		}
+		fapi.AppendQanAPIInstancesId(agentInstance.UUID, agentInstance)
+		fapi.Append(fmt.Sprintf("/qan-api/agents/%s/cmd", agentInstance.UUID), func(w http.ResponseWriter, r *http.Request) {
+			switch r.Method {
+			case "PUT":
+				body, err := ioutil.ReadAll(r.Body)
+				if err != nil {
+					panic(fmt.Sprintf("error reading body: %s", err))
+				}
+				defer r.Body.Close()
+
+				cmd := proto.Cmd{}
+				err = json.Unmarshal(body, &cmd)
+				if err != nil {
+					panic(fmt.Sprintf("error unmarshaling body: %s", err))
+				}
+				err = json.Unmarshal(cmd.Data, &config)
+				if err != nil {
+					panic(fmt.Sprintf("error unmarshaling body: %s", err))
+				}
+
+				w.WriteHeader(http.StatusOK)
+			default:
+				w.WriteHeader(600)
```

Comment:
600?

Reply:
I just did it to indicate this is unsupported call, in contrast to returning 500 which could be a real response. Maybe `panic` would be better here. What do you think?